packages/web3-eth-abi/src/utils.ts

```diff
@@ -0,0 +1,250 @@
+import { AbiCoder, ParamType } from '@ethersproject/abi';
+import { leftPad, rightPad, toHex } from 'web3-utils';
+import ethersAbiCoder from './ethers_abi_coder';
+import {
+	AbiInput,
+	JsonAbiCoderStruct,
+	JsonAbiFragment,
+	JsonAbiParameter,
+	JsonAbiStruct,
+	JsonAbiEventFragment,
+	JsonAbiFunctionFragment,
+} from './types';
+
+export const isAbiFragment = (item: unknown): item is JsonAbiFragment =>
+	item !== undefined &&
+	item !== null &&
+	typeof item === 'object' &&
+	(item as { type: string }).type !== undefined &&
+	['function', 'event', 'constructor'].includes((item as { type: string }).type);
```

Comment:
I think more good to check against type defined in types.ts `type FragmentTypes = 'constructor' | 'event' | 'function';`
so future developers dnt miss if making changes

Reply:
Those defined there are types and can't be used as values for validation.